File: Easy/1337. The K Weakest Rows in a Matrix/1337. The K Weakest Rows in a Matrix.java
class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        int[] result = new int[k];
        int m = mat.length;
        int n = mat[0].length;
        int[] strengths = new int[m];

        // Calculate the strength of each row
        for (int i = 0; i < m; i++) {
            int strength = 0;
            for (int j = 0; j < n; j++) {
                strength += mat[i][j];
            }
            strengths[i] = strength * 1000 + i; // Use a multiplier to preserve original index
        }

        // Sort the rows by strength and index in case of a tie
        Arrays.sort(strengths);

        // Extract the indices of the k weakest rows
        for (int i = 0; i < k; i++) {
            result[i] = strengths[i] % 1000; // Retrieve the original index
        }

        return result;
    }
}

File: Easy/1337. The K Weakest Rows in a Matrix/1337. The K Weakest Rows in a Matrix.rs
use std::collections::BinaryHeap;

impl Solution {
    pub fn k_weakest_rows(mat: Vec<Vec<i32>>, k: i32) -> Vec<i32> {
        let mut heap: BinaryHeap<(i32, i32)> = BinaryHeap::new(); // Explicit type annotation

        for (i, row) in mat.iter().enumerate() {
            let strength = row.iter().sum();
            heap.push((strength, i as i32));
            if heap.len() > k as usize {
                heap.pop();
            }
        }

        let mut result = Vec::with_capacity(k as usize);
        while let Some((_, i)) = heap.pop() {
            result.push(i);
        }

        result.reverse(); // Reverse the result to get the weakest rows first
        result
    }
}
